buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -304,13 +304,15 @@ class ClusterFormationTasks {
     static Task configureWriteConfigTask(String name, Project project, Task setup, NodeInfo node, NodeInfo seedNode) {
         Map esConfig = [
                 'cluster.name'                 : node.clusterName,
-                'node.name'                    : "node-" + node.nodeNum,
                 'pidfile'                      : node.pidFile,
                 'path.repo'                    : "${node.sharedDir}/repo",
                 'path.shared_data'             : "${node.sharedDir}/",
                 // Define a node attribute so we can test that it exists
                 'node.attr.testattr'           : 'test'
         ]
+        if (node.config.addStandardNodeName) {
```

Comment:
Could we do this without any new setting by changing the injection of settings to remove/omit when the value is null? This would then generally allow any settings to be removed.

Reply:
I think that sounds more confusing then adding the setting. I mean, it is more flexible but I don't know that we need that flexibility.

Reply:
Fewer settings on NodeConfiguration is better, IMO. It makes less logic on node setup.